### ISSUE TITLE
Not-a-numbers do not always follow IEEE semantics

### ISSUE THREAD

**Issue by jcbrinfo** [nitlang/nit#2314]
```
$ ../bin/nit -e '
> print(0.0/0.0 != 0.0/0.0)
> var x = 0.0/0.0
> print(x.is_nan)
> print(x != x)
> print(x != x * 1.0)'
true
true
false
true
```

**Comment by privat** [comment]
Yeah, there is a choice to make here. Either Object::== is the opposite of Object::!= and does not contradict Object::is_same_instance, that are some strong laws needed by a lot of data-structures and algorithms, either we follow the crappy IEEE semantic on NaN.

By the way, other OO language seems have such a hard time. Even Java https://docs.oracle.com/javase/7/docs/api/java/lang/Double.html#equals(java.lang.Object)

Nevertheless, consistency need to be achieved in Nit. The underlying problem here seems that == and != are implemented by something like: compare the instance boxe addresses, if different do a == in C.
